hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/token/delegation/AbstractDelegationTokenSecretManager.java

```diff
@@ -64,7 +69,13 @@ private String formatTokenId(TokenIdent id) {
    */
   protected final Map<TokenIdent, DelegationTokenInformation> currentTokens 
       = new ConcurrentHashMap<>();
-  
+
+  /**
+   * Map of token real owners to its token count. This is used to generate
+   * top users by owned tokens.
+   */
+  protected final Map<String, Long> tokenOwnerStats = new ConcurrentHashMap<>();
```

Comment:
Seems this won't be updated if a standby NN updates its own token info by pulling from edit log?

Reply:
It depends on individual secret manager to initialize the `currentTokens`. In namenode, it is loading from edit log. In router, it is loading from ZK.
I would have a separate ticket for namenode.

Reply:
OK. Can you add a comment for this though indicating that this only support RBF for now?